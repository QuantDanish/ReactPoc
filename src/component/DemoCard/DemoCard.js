import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  index: PropTypes.number,
  title: PropTypes.string,
  minWidth: PropTypes.string
}

const defaultProps = {
  children: 'Dummy Child',
  title: 'Title',
  index: 0,
  minWidth: '48%'
}

const DemoCard = ({children, index, title, minWidth}) => {
  return (
    <div className="card p-2 m-2" style={{width: minWidth}}>
      <div className="h4">
        {index}. {title}
      </div>
      <hr/>
      <div className="card-body">
        {children}
      </div>
    </div>
  );
}


DemoCard.propTypes = propTypes;
DemoCard.defaultProps = defaultProps;


export default DemoCard;
