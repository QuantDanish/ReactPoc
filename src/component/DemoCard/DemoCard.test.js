import React from 'react';
import {configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import DemoCard from './DemoCard';

configure({adapter: new Adapter()});

describe('Demo Component Testing', () => {
  it('Should renders without crashing', () => {
    let component = shallow(<DemoCard></DemoCard>)
    expect(component).toMatchSnapshot();
  });

  it('Should renders without crashing with props', () => {
    let component = shallow(<DemoCard index={1} title="title" minWidth="50%">Text</DemoCard>);
    expect(component).toMatchSnapshot();
  });

  // it('Should matches the given prop', () => {
  //   let component = shallow(<DemoCard index={1}></DemoCard>);
  //   const recieved = component.props().index;
  //   console.log(component.props());
  //   expect(recieved).toBe(1);
  // })
  
})

