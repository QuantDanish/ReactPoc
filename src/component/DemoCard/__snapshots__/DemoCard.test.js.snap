// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo Component Testing Should renders without crashing 1`] = `
<div
  className="card p-2 m-2"
  style={
    Object {
      "width": "48%",
    }
  }
>
  <div
    className="h4"
  >
    0
    . 
    Title
  </div>
  <hr />
  <div
    className="card-body"
  >
    Dummy Child
  </div>
</div>
`;

exports[`Demo Component Testing Should renders without crashing with props 1`] = `
<div
  className="card p-2 m-2"
  style={
    Object {
      "width": "50%",
    }
  }
>
  <div
    className="h4"
  >
    1
    . 
    title
  </div>
  <hr />
  <div
    className="card-body"
  >
    Text
  </div>
</div>
`;
